src/app/ToDo/Components/to-do.component.ts

```diff
@@ -1,66 +1,92 @@
 import { Component, OnInit } from '@angular/core';
+import {
+  TodosState
+} from '../state'
+import * as TodoActions from '../state/todo.action'
+import { ToDo } from '../state/todo.model';
+import { Store } from '@ngrx/store';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ToDoFacade } from '../state/todo.facade';
 import { Observable, Subscription } from 'rxjs';
-import { map } from 'rxjs/operators';
-import ToDo from '../state/todo.model';
-import ToDoState from '../state/todo.state';
-import { ToDoFacade } from './../state/todo.facade';
 
 @Component({
   selector: 'app-to-do',
-  templateUrl: './to-do.component.html',
-  styleUrls: ['./to-do.component.css']
+  templateUrl: './to-do.component.html'
 })
 export class ToDoComponent implements OnInit {
-  todo$: Observable<ToDoState>;
-  ToDoSubscription: Subscription;
-  ToDoList: ToDo[] = [];
 
-  Title: string = '';
-  IsCompleted: boolean = false;
-  todoError: Error = null;
+  public todo$: Observable<any>;
+  public todoForm: FormGroup;
+  public toDoSubscription: Subscription;
 
-  constructor(private _todoFacade: ToDoFacade) {
+  public loadTask(task: ToDo) {
+    this.todoForm.patchValue(task);
+  }
+
+  constructor(
+    private _todoFacade: ToDoFacade,
+    private _store: Store<TodosState>,
+    private _formBuilder: FormBuilder) {
     this.todo$ = _todoFacade.todo$;
   }
 
-  ngOnInit() {
-    
-    //Calling the facade method, to give all tasks
-    this._todoFacade.getAllTasks();
+  public ngOnInit() {
+    this._todoFacade.getTasks();
+
+    this.buildForm();
+  }
+
+  buildForm() {
+    this.todoForm = this._formBuilder.group({
+      id: [null],
+      text: [, Validators.compose([Validators.required])],
+      completed: [false],
+    });
+  }
+
+  /**
+   * Adding or editing a task
+   */
+  public async saveTodo() {
+
+    //check if form is valid
+    if (!this.todoForm.valid) return;
+
+    var todo: ToDo = this.todoForm.value;
 
-    //Listening this observable, so that brings  error or tasks available
-    this.ToDoSubscription = this.todo$
-      .pipe(
-        map(x => {
-          this.ToDoList = x.ToDos;
-          this.todoError = x.ToDoError;
-        })
-      )
-      .subscribe();
+    if (todo.id)
+      await this._todoFacade.editTask(todo);
+    else {
+      todo = new ToDo(todo.text, todo.completed);
+      await this._todoFacade.newTask(todo);
+    }
+
+    //reseting fields
+    this.todoForm.reset();
   }
 
   /**
-   * Call facade method responsible to create task
+   * Marking task with check or un-check
+   * @param id 
```

Comment:
We do not use `@param` in our projects.

Try to write the names of methods and variables so that they do not require a comment :)

Reply:
Cool !